mono/mini/interp/interp.c

```diff
@@ -5279,15 +5283,16 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, FrameClause
 		}
 		MINT_IN_CASE(MINT_LDELEMA)
 		MINT_IN_CASE(MINT_LDELEMA_TC) {
-			gboolean needs_typecheck = *ip == MINT_LDELEMA_TC;
 			
-			MonoClass *klass = (MonoClass*)imethod->data_items [*(guint16 *) (ip + 1)];
 			guint16 numargs = *(guint16 *) (ip + 2);
 			ip += 3;
 			sp -= numargs;
 
 			o = sp [0].data.o;
 			NULL_CHECK (o);
+
+			MonoClass *klass = (MonoClass*)imethod->data_items [*(guint16 *) (ip - 3 + 1)];
+			const gboolean needs_typecheck = ip [-3] == MINT_LDELEMA_TC;
```

Comment:
For example these should not matter because they are only after a null check -- if there is a function call, it get cannot get here. I'm not sure the compiler was being that good.